core/groupby.cpp

```diff
@@ -197,20 +195,24 @@ template <class T> struct MakeGroups {
     const auto &values = key.values<T>();
 
     const auto dim = key.dims().inner();
-    std::map<T, std::vector<Slice>> indices;
-    for (scipp::index i = 0; i < scipp::size(values);) {
+    std::map<T, GroupByGrouping::group> indices;
+    const auto end = values.end();
+    scipp::index i = 0;
+    for (auto it = values.begin(); it != end;) {
```

Comment:
In generic code here and below looks like you could potentially elide some copies. Might be significant depending on `T`.

Reply:
Can you clarify?

Reply:
Oh I see, you mean, e.g., when `T` is `std::string`?

Reply:
Yes, for example `auto v = *it` or similar.

Reply:
Should be fixed in the update I pushed a couple minutes ago.

Reply:
Yes looks good.